src/python/WMCore/ReqMgr/Utils/Validation.py

```diff
@@ -63,15 +64,31 @@ def validate_request_update_args(request_args, config, reqmgr_db_service, param)
         elif request_args["RequestStatus"] == 'assigned':
             workload.validateArgumentForAssignment(request_args)
 
-    # TODO: fetch it from the assignment arg definition
-    if 'RequestPriority' in request_args:
-        request_args['RequestPriority'] = int(request_args['RequestPriority'])
-        if (lambda x: (x >= 0 and x < 1e6))(request_args['RequestPriority']) is False:
-            raise InvalidSpecParameterValue("RequestPriority must be an integer between 0 and 1e6")
+    validate_request_priority(request_args)
 
     return workload, request_args
 
 
+def validate_request_priority(reqArgs):
+    """
+    Validate the RequestPriority argument against its definition
+    in StdBase
+    :param reqArgs: dictionary of user request arguments
+    :return: nothing, but raises an exception in case of an invalid value
+    """
+    if 'RequestPriority' in reqArgs:
+        reqPrioDefin = StdBase.getWorkloadCreateArgs()['RequestPriority']
+        try:
+            # Web GUI sends it as a string
+            reqArgs['RequestPriority'] = int(reqArgs['RequestPriority'])
```

Comment:
Wouldn't that be more clear if one just does the following here:

```
if not isinstance(reqArgs['RequestPriority'], reqPrioDefin['type']):
    raise InvalidSpecParameterValue(msg)
```
We need to check for the type only and not for the possibility to cast the object to a given type, because we never do so down the code. And we will also avoid hardcoding a `int` type check in the middle of the code while we have the object type defined from StdBase. I think we should take it from the upper level object otherwise we risk to have objects that silently transform from one type to another, which in my opinion is a bug.

Reply:
Good catch. I meant to write:
```
reqArgs['RequestPriority'] = reqPrioDefin['type'](reqArgs['RequestPriority'])
```

About simply checking the data type, it would be easier if we were already running this software in python3. Since we are still in py2, any check for string is just another complication in the near future.